spacy/cli/debug_data.py

```diff
@@ -397,3 +547,18 @@ def _get_labels_from_model(nlp, pipe_name):
         return set()
     pipe = nlp.get_pipe(pipe_name)
     return pipe.labels
+
+
+def _get_nonprojective_examples(data):
```

Comment:
Can we move some of these into the `_compile_gold` helper? I've been trying to limit the times we loop over the entire data and for overall counts, I do think we can take care of all of this upfront. (It won't work for cases like `_get_examples_without_label`, though, because that needs to run for specific labels.)

That said, speed definitely isn't a big concern here and it's fine if the user has to wait a little to have their whole corpus analysed.

Reply:
Sure, and some of the things it's counting can be removed, since they really shouldn't happen. (Does projectivize introduce cycles? Only if spacy has done something really wrong, but I wanted to rule it out for the Chinese corpus discussion.)